chart/templates/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "canary-checker.name" . }}
+  labels:
+    {{- include "canary-checker.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      {{- include "canary-checker.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "canary-checker.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "canary-checker.name" . }}-sa
+      volumes:
+        - name: dockersock
```

Comment:
Can we gate this behind a .Values.dockerSocket = true` ?

Reply:
I've gated the volume creation and mounting of that volume.